scylla-cdc-base/src/main/java/com/scylladb/cdc/model/worker/WorkerConfiguration.java

```diff
@@ -72,11 +72,27 @@ public Builder withCQL(WorkerCQL cql) {
             return this;
         }
 
-        public Builder withConsumer(TaskAndRawChangeConsumer consumer) {
-            this.consumer = Preconditions.checkNotNull(consumer);
+        public Builder withConsumer(Consumer consumer) {
+            this.consumer = consumer;
             return this;
         }
 
+        public Builder withConsumer(TaskAndRawChangeConsumer consumer) {
+            return withTaskAndRawChangeConsumer(consumer);
+        }
+
+        public Builder withTaskAndRawChangeConsumer(TaskAndRawChangeConsumer consumer) {
+            return withConsumer(Consumer.forTaskAndRawChangeConsumer(Preconditions.checkNotNull(consumer)));
+        }
+
+        public Builder withConsumer(RawChangeConsumer consumer) {
+            return withRawChangeConsumer(consumer);
+        }
+
+        public Builder withRawChangeConsumer(RawChangeConsumer consumer) {
+            return withConsumer(Consumer.forRawChangeConsumer(Preconditions.checkNotNull(consumer)));
+        }
+
```

Comment:
I don't understand why we need both withConsumer and withTaskAndRawChangeConsumer. Same with withConsumer and withRawChangeConsumer. Can't we just have two overloads of withConsumer?

Reply:
Lambdas. With `withConsumer` overloads the compiler does not know which overload to use generally, when using a lambda callback.

Reply:
Then let's drop withConsumer or is there a reason not to?

Reply:
Well, overloads are useful when you _do_ have concrete typed objects, i.e. for example a generic type calling. Being a fan of usability, I like having the various options.